encodings/protobuf/README.md

```diff
@@ -0,0 +1,178 @@
+# OCSF Protobuf Encoding
+
+This directory contains a sample Protobuf encoding for OCSF.
+
+Although encoding in general is _informative_ in OCSF, enabling the
+interoperation between systems in a single organization and indeed
+between organizations indicates that a **canonical** protobuf encoding
+has significant utility.  This directory contains that canonical
+encoding.
+
+## Semantic Type Considerations
+
+### Primitive Types
+
+The primitive types are mapped as follows
+
+| OCSF Type   | Protobuf Type           |
+| ----------- | ----------------------- |
+| `boolean_t` | `bool`                  |
+| `float_t`   | `float`                 |
+| `integer_t` | `int32`                 |
+| `long_t`    | `int64`                 |
+| `string_t`  | `string`                |
+| `json_t`    | `google.protobuf.Value` |
+
+The mapping is straightforward, except to note that `json_t` and the
+bare (un-extended) `object` type are represented by well-known
+protobuf types that model JSON, specifically
+[Value](https://protobuf.dev/reference/protobuf/google.protobuf/#value)
+and
+[Struct](https://protobuf.dev/reference/protobuf/google.protobuf/#struct)
+respectively.
+
+### Optionality
+
+Most attributes in OCSF are not required; for non-array primitive
+types, the corresponding protobuf descriptor will mark them explicity
+as `optional`.  This has implications for most SDKs generated by
+`protoc`.  For example, in Go, the field of the struct becomes of
+pointer type, adding an extra indirection with the benefit that the
+presence or absence of the field can be consistently determined.
+
+Required attributes in OCSF are not marked as optional.  In most
+cases, an SDK will still permit distinguishing between present and
+absent values because the zero value is not a valid value anyway.  For
+example, `class_id` is required and non-zero; if the application finds
+it to have a zero value, the distinction between missing and invalid
+is not significant because missing _is_ invalid for required types.
+
```

Comment:
It may be worth noting that the OCSF schema does not distinguish between missing and `null` / `nil` values -- both can be considered equivalent. (A perhaps helpful distinction.)